carbon-design-system/carbon
# fix(16740): End Date calendar in range variant remains open on Click away

Closes #https://github.com/carbon-design-system/carbon/issues/16740

This PR fixes the End Date calendar in range variant remains open on Click away 

#### Changelog
```
 // overriding the flatpickr bug where the startDate gets deleted on blur
if (inputTo === target && !inputFrom.value && lastStartValue.current) {
      if (isValidDate(parseDateWithFormat(lastStartValue.current))) {
        inputFrom.value = lastStartValue.current;
        if (inputTo.value) {
          fp.setDate(
            [inputFrom.value, inputTo.value],
            false,
            fp.config.dateFormat
          );
          fp.updateValue();
        }
      }
    }
```



For [setDate](https://flatpickr.js.org/instance-methods-properties-elements/#setdatedate-triggerchange-datestrformat) function :  `false`  is passed as second argument to avoid forcing any onChange .

`updateValue` is invoked to maintain the selected range (blur color on selected range) when the calendar is opened again.

https://github.com/carbon-design-system/carbon/assets/63502271/e4ff095f-b203-47f6-a163-c80a45575924


#### Testing / Reviewing
1. Open datepicker with a range
2. Select start and end date
3. Open end date calendar, do not select any date and click outside of calendar and verify if calendar closes
4. Made sure the selected range is visible when the  calendar is opned again 
5. Verify existing functionality is intact
